lib/sles4sap_publiccloud.pm

```diff
@@ -496,65 +494,113 @@ sub cleanup_resource {
+    my $retry_count = 0;    # counter of retries
+    my $vhost;    # hostname from 'vhost' value of 'Host' key of SAPHanaSR-showAttr topology
+    my $sync_state;    # status of the sync from 'srPoll' value of 'Site' key of SAPHanaSR-showAttr topology
 
   TAKEOVER_LOOP: while (1) {
         my $topology = $self->get_hana_topology();
         $retry_count++;
-        while (my ($entry, $host_entry) = each %$topology) {
-            foreach (qw(vhost sync_state)) {
-                die("Missing '$_' field in topology output") unless defined(%$host_entry{$_}); }
-            my $vhost = %$host_entry{vhost};
-            my $sync_state = %$host_entry{sync_state};
-            record_info("Cluster Host", join("\n",
+        for my $site (keys %{$topology->{Site}}) {
+            die("check_takeover [ERROR] Missing 'srPoll' field in topology output of Site->$site")
+              unless defined($topology->{Site}->{$site}->{srPoll});
+            # As 'vhost' value is not part of the 'Site' key we need to pair it with corect 'Host' key
+            for my $host (keys %{$topology->{Host}}) {
+                die("check_takeover [ERROR] Missing 'vhost' field in topology output of $host")
+                  unless defined($topology->{Host}->{$host}->{vhost});
+                $vhost = $topology->{Host}->{$host}->{vhost} if ($topology->{Host}->{$host}->{site} eq $site);
+                $sync_state = $topology->{Site}->{$site}->{srPoll} if ($topology->{Host}->{$host}->{site} eq $site);
+            }
+            record_info('Cluster Host', join("\n",
                     "vhost: $vhost compared with $hostname",
                     "sync_state: $sync_state compared with PRIM"));
             if ($vhost ne $hostname && $sync_state eq "PRIM") {
-                record_info("Takeover status:", "Takeover complete to node '$vhost");
+                record_info('Takeover status:', "Takeover complete to node '$vhost'");
                 last TAKEOVER_LOOP;
             }
         }
-        die("Test failed: takeover failed to complete.") if ($retry_count > 40);
+        die('check_takeover [ERROR] Test failed: takeover failed to complete.') if ($retry_count > 40);
         sleep 30;
     }
 
     return 1;
 }
 
 =head2 enable_replication
-    enable_replication([site_name => 'site_a']);
+    enable_replication([site_name => 'site_a', hana_inn => '10' ]);
 
-    Enables replication on fenced database. Database needs to be offline.
+    Re-enables replication on the previously fenced database node. Database must be offline.
+<<<<<<< HEAD
+    Dies if the the given <site name> doesn't correspond with the SAPHanaSR topology local site name
+    Optionally the hana_instance_id could be provided for better readability
+=======
+    Dies if the the given <site name> doesn't corespond with the SAPHanaSR topology of local site.
+    Optionaly the 'hana_innstance_id' could be provided for better reliability
+>>>>>>> 222157187 (More comments)
```

Comment:
This looks like a left over from a manual `git rebase`. Please fix.

Reply:
Will remove